sisyphus/job.py

```diff
@@ -289,17 +289,18 @@ def _sis_setup_directory(self, force=False):
             pickle.dump(self, f)
 
         with open(self._sis_path(gs.JOB_INFO), "w", encoding="utf-8") as f:
-            for tag in self.tags:
+            for tag in sorted(str(t) for t in self.tags):
```

Comment:
The main problem is that `self.tags` (`self._sis_tags`), `self._sis_inputs` and `self._sis_aliases` are `set`s, and the order of a `set` is non-deterministic and arbitrary. We could also replace `set` by some similar structure which keep a well defined order (insertion order). Or otherwise `sorted` is also fine, definitely better than the current code.

Reply:
I see value in both options. Personally, I prefer `sorted` because it allows you to binary search w/ your eyes when you are looking for a specific value.